src/structured_config.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Structured configuration for the Superset charm."""
+import logging
+from enum import Enum
+from typing import Optional
+
+from charms.data_platform_libs.v0.data_models import BaseConfigModel
+from pydantic import validator
+
+logger = logging.getLogger(__name__)
+
+
+class BaseEnumStr(str, Enum):
+    """Base class for string enum."""
+
+    def __str__(self) -> str:
+        """Return the value as a string.
+
+        Returns:
+            string of config value
+        """
+        return str(self.value)
+
+
+class LogLevelType(str, Enum):
```

Comment:
question: Did you mean to use `BaseEnumStr` as the parent in these classes?

Reply:
You're right. I've actually opted to remove the `BaseEnumStr` class altogether as there seems to be a limitation in one of the charm classes that does not represent the enum value properly when accessed without `.value`.